### ISSUE TITLE
Repo manager accepts blank name

### ISSUE THREAD

**Issue by david4096** [ga4gh/server#1211]
Although checks are in place to keep the inferred names from being empty, it appears one can manually set the name to be empty.

`python repo_dev.py add-readgroupset l a ga4gh-example-data/HG00096.bam --name ""`

**Comment by jeromekelleher** [comment]
Good catch. We should add a method to SqlDataRepository called ``checkNameFormat`` (or something similar) that is called during the ``insertX`` methods.

**Comment by dcolligan** [comment]
This could probably use some design up-front.  We're basically talking about adding a object validation layer before DB insertion.

In fact it might be good to refactor the data repo insertion code to make it more OO.  We could have some logic like:

```
errors = object.validate()
if not errors.isEmpty():
    sql = object.getSql()
    args = object.getArguments()
    try:
        cursor.execute(sql, args)
    except whateverException as exception:
        object.handleInsertionException(exception)
else:
    raise dbObjectValidationFailedException(object, errors)
```

Or some variation of that in the datarepo code.  Then we'd have an object hierarchy of `DbObject` where each subclass would implement `validate`, `getSql`, etc.

Thoughts?

**Comment by jeromekelleher** [comment]
This looks like a nice pattern @dcolligan, and I agree that we should try to add more structure to the admin process soon. However, I would like to build out the admin features more before we stand back and refactor/redesign things. The code is still reasonably clear, and there's a lot of features yet to go in. I'm wary of trying to do too much design before the full feature set has been fleshed out.

Plus, there's a big refactor in the pipelines changing the object ID structure, as well as using SQL to directly answer queries. I think it would be good to get these out of the way before we return to the insertion logic design.

How does this sound?

**Comment by dcolligan** [comment]
It's not a solution I'm thrilled about, but if you think a refactor would hamper existing/future efforts, then I suppose I can pepper the code with `checkNameFormat` calls...
